test/components/syncTest.js

```diff
@@ -0,0 +1,788 @@
+    const deletedTitle = this.page3Title
+    yield Brave.app.client
+      .waitUntil(function () {
+        return this.getText(bookmarksToolbar)
+          .then((allBookmarks) => allBookmarks.includes(deletedTitle) === false)
+      })
+  })
+
+  it('create folder', function * () {
+    const title = this.folder1Title
+    yield Brave.app.client
+      .waitUntil(function () {
+        return this.getText(bookmarksToolbar)
+          .then((allBookmarks) => allBookmarks.includes(title))
+      })
+  })
+
+  it('create bookmark in folder', function * () {
+    const pageNthChild = 1
+    const folderTitle = this.folder1Title
+    const pageTitle = this.folder1Page1Title
+    const folder = `.bookmarkToolbarButton[title="${folderTitle}"]`
+    yield Brave.app.client
+      .waitForVisible(folder)
+      .click(folder)
+      .waitForVisible('.contextMenu')
+      .waitUntil(function () {
+        return this.getText(`.contextMenuItem:nth-child(${pageNthChild})`)
+          .then((bookmark1Title) => bookmark1Title === pageTitle)
+      })
+  })
+
+  it('update bookmark, moving it into the folder', function * () {
+    const pageNthChild = 2
+    const folderTitle = this.folder1Title
+    const pageTitle = this.folder1Page2Title
+    const folder = `.bookmarkToolbarButton[title="${folderTitle}"]`
+    yield Brave.app.client
+      .waitForVisible(folder)
+      .click(folder)
+      .waitForVisible('.contextMenu')
+      .waitUntil(function () {
+        return this.getText(`.contextMenuItem:nth-child(${pageNthChild})`)
+          .then((bookmark1Title) => bookmark1Title === pageTitle)
+      })
+  })
+
+  it('delete folder', function * () {
+    const title = this.folder2Title
+    yield Brave.app.client
+      .waitUntil(function () {
+        return this.getText(bookmarksToolbar)
+          .then((allBookmarks) => allBookmarks.includes(title) === false)
+      })
+  })
+
+  it('delete folder deletes pages within', function * () {
+    const title = this.folder2Page1Title
+    yield Brave.app.client
+      .waitUntil(function () {
+        return this.getText(bookmarksToolbar)
+          .then((allBookmarks) => allBookmarks.includes(title) === false)
+      })
+  })
+
+  it('sync order', function * () {
+    const pageTitle = this.page1Title
+    const updatedTitle = this.page2TitleUpdated
+    const folder1Title = this.folder1Title
+
+    yield Brave.app.client
+      .waitUntil(function () {
+        return this.getText('.bookmarkToolbarButton:nth-child(1) .bookmarkText')
+          .then((title) => title === pageTitle)
+      })
+      .waitUntil(function () {
+        return this.getText('.bookmarkToolbarButton:nth-child(2) .bookmarkText')
+          .then((title) => title === updatedTitle)
+      })
+      .waitUntil(function () {
+        return this.getText('.bookmarkToolbarButton:nth-child(3) .bookmarkText')
+          .then((title) => title === folder1Title)
+      })
+  })
+})
+
+describe('Syncing history', function () {
```

Comment:
not sure what you mean by a 'tie'. if the same history entry (same location) has two sync records with the same timestamp, the one which is received later by the sync server wins.